portfolio/src/main/webapp/script.js

```diff
@@ -55,20 +55,42 @@ function addRandomDestination() {
 /**
  * adds response from servlet
  */
+function setComments() {
+  num_comments = document.getElementById('num-comments').value;
+  order = document.getElementById('order').value;
+  fetch(
+      '/data' +
+      '?num-comments=' + num_comments + '&order=' + order)
+      .then((response) => response.json())
```

Comment:
Are you familiar with `async` functions? This might be easier to to read if you convert the chained futures into an imperative `async/await` sequence.

I'll leave it up to you to choose whether you want to do so, but here's an intro to [async functions](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/async_function) in any case.